test/neo/client/request/internal/GetAll.d

```diff
@@ -213,8 +181,178 @@ public struct GetAll
+
+    /// Enum of signals used by the request fibers.
+    private enum Signals
+    {
+        Stop = 1
+    }
+
+    /// Fiber which handles iterating and sending records to the client.
+    private class Reader
+    {
+        private MessageFiber fiber;
+
+        public this ( )
+        {
+            this.fiber = new MessageFiber(&this.fiberMethod, 64 * 1024);
+        }
+
+        void fiberMethod ( )
+        {
+            uint count;
+            bool finished;
+            do
+            {
+                auto message = this.outer.request_event_dispatcher.receive(this.fiber,
+                    Message(MessageType.Record), Message(MessageType.End));
+                with ( MessageType ) switch ( message.type )
+                {
+                    case Record:
+                        count++;
+
+                        auto key = *conn.message_parser.
+                            getValue!(hash_t)(message.payload);
+                        auto value = conn.message_parser.
+                            getArray!(char)(message.payload);
+
+                        GetAll.Notification n;
+                        n.record = RequestKeyDataInfo(context.request_id,
+                            key, value);
+                        GetAll.notify(context.user_params, n);
+                        break;
+
+                    case End:
+                        finished = true;
+                        break;
+
+                    default:
+                        assert(false);
+                }
+
+                // Stop the GetAll after some records have been received.
+                // In a real client, the request would only be stopped if
+                // requested via the user-facing API. In this example, for
+                // simplicity, the Reader triggers a Stop message.
+                if ( count == 5 )
+                    this.outer.request_event_dispatcher.signal(this.outer.conn,
+                        Signals.Stop);
+            }
+            while ( !finished );
+
+            // Kill the controller fiber.
+            this.outer.request_event_dispatcher.abort(
+                this.outer.controller.fiber);
+        }
+    }
+
+    /// Fiber which handles control messages from the client.
+    private class Controller
+    {
+        MessageFiber fiber;
+
+        this ( )
+        {
+            this.fiber = new MessageFiber(&this.fiberMethod, 64 * 1024);
+        }
+
+        void fiberMethod ( )
+        {
+            // Wait for a control message to be initiated.
+            // In a real client, there would be a user-facing API to do this. In
+            // this example, for simplicity, the Reader triggers a control
+            // message.
+            auto event = this.outer.request_event_dispatcher.nextEvent(
+                this.fiber, Signal(Signals.Stop));
+            assert(event.active == event.active.signal);
+            assert(event.signal.code == Signals.Stop);
```

Comment:
The `in` contract of `event.signal()` contains ` assert(event.active == event.active.signal)`.